utils/ensmallen_utils.hpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file utils.hpp
+ * @author Kartik Dutt
+ *
+ * Definition of Ensmallen Utility functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef ENSMALLEN_CALLBACKS_PRINT_METRIC_HPP
+#define ENSMALLEN_CALLBACKS_PRINT_METRIC_HPP
+
+#include <ensmallen.hpp>
+#include <functional>
+
+namespace ens {
+
+/**
+ * Prints metric on validation set.
+ */
+template<typename AnnType,
+         class MetricType,
+         typename InputType = arma::mat,
+         typename OutputType = arma::mat
+>
+class PrintMetric
```

Comment:
Kindly let me know if this would be a good addition to ensmallen as well? I can open a PR for it there.

Reply:
https://github.com/mlpack/ensmallen/issues/200
I am going to resolve this for now.